src/qvk/qvk.c

```diff
@@ -702,11 +702,27 @@ qvk_init()
   };
 
   /* create device and queue */
-  QVK(vkCreateDevice(qvk.physical_device, &dev_create_info, NULL, &qvk.device));
+  int res = vkCreateDevice(qvk.physical_device, &dev_create_info, NULL, &qvk.device);
```

Comment:
for this lengthy dance, there's the QVKR() macro

Reply:
From experience it is a bad idea to do a return in a macro. We had trouble in Samba with this and don't allow it anymore. But I can do that if you want.

Reply:
a) is the return value in the macro the correct one for all situation where the macro is called?
b) someone might not assume that it returns

Reply:
replace one line of code with 5 same lines of boilerplate code in many places? i don't think that helps legibility. isn't code duplication also one of the things that are written in books to not do? (if you're worried about this macro you'd love the doubly linked list..)